tests/components/hassio/test_sensor.py

```diff
@@ -303,22 +305,31 @@ async def test_stats_addon_sensor(
     _install_default_mocks(aioclient_mock)
     _install_test_addon_stats_mock(aioclient_mock)
 
-    async_fire_time_changed(
-        hass, dt_util.utcnow() + HASSIO_UPDATE_INTERVAL + timedelta(seconds=1)
-    )
+    freezer.tick(HASSIO_UPDATE_INTERVAL + timedelta(seconds=1))
+    async_fire_time_changed(hass)
     await hass.async_block_till_done()
 
-    # Enable the entity.
+    assert "Could not fetch stats" not in caplog.text
+
+    # Enable the entity and wait for the reload to complete.
     entity_registry.async_update_entity(entity_id, disabled_by=None)
-    await hass.config_entries.async_reload(config_entry.entry_id)
+    freezer.tick(config_entries.RELOAD_AFTER_UPDATE_DELAY)
+    async_fire_time_changed(hass)
     await hass.async_block_till_done()
+    assert config_entry.state is config_entries.ConfigEntryState.LOADED
```

Comment:
This test only worked because the coordinator refresh wasn't cancelled at reload before